### DISCUSSION TITLE
Expanding Window Splitter and Refitting

### DISCUSSION THREAD

**Discussion by edzWH** [alan-turing-institute/sktime#5704] [Q&A] [answered]
Hi all, 

I'm just starting out with SkTime and have a question about CV with the Expanding Window Splitter within a random grid search.

If Refit=True in Randomised Search, wouldn't possibly lead to overfitting? For instance, say if I have 5 different train/test splits across the dataset, refitting the data each time will select the best model from any of those splits (I presume). Where as I'd prefer to get the model which has the highest average score across all of the CV splits to prevent overfitting to 1 test set. 

As a example, here is some manual cross validation visualisation I've done previously:

<img width="616" alt="traintestsplit" src="https://github.com/sktime/sktime/assets/155727279/4ed9466b-7f2d-4953-9b6b-9572eecab1be">

Ideally I would like the selected model to be the highest average score across all of those sets, is this possible to do within SkTime?

Is expanding window splitter the correct validation type for what I'd like to do?

This is likely my misunderstanding of the library as I'm generally new to ML, any help would be appreciated. 

Thanks.

**Comment by fkiraly** [comment, marked as answer]
@edzWH, not sure if I understand the question - `refit` fits a new model from scratch, five times, and then the average across five folds is computed, this average is used to select the best parameters.

Can you describe in detail what you would like to happen instead?

**Reply by edzWH** [reply]
This answers my question thanks, my misunderstanding on some of the terminology, appreciate your help

**Reply by fkiraly** [reply]
Thanks - I've updated the docstring based on our small exchage, hopefully this improves clarity of what `"refit"` does: https://github.com/sktime/sktime/pull/5711
